### PR TITLE
Hunterised tinyxml2.

### PR DESCRIPTION
**Pull request by lsolanka**
Replace absolute installation paths with relative.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -173,4 +173,4 @@ install(FILES
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${CMAKE_PROJECT_NAME})
 
 install(EXPORT ${CMAKE_PROJECT_NAME}Targets
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${CMAKE_PROJECT_NAME})
```
Comment (ruslo) [inline CMakeLists.txt]: `CMAKE_INSTALL_LIBDIR` should work fine actually. It's a relative path, see:
* https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -60,7 +60,7 @@ set_target_properties(tinyxml2 PROPERTIES
 if(DEFINED CMAKE_VERSION AND NOT "${CMAKE_VERSION}" VERSION_LESS "2.8.11")
     target_include_directories(tinyxml2 PUBLIC 
                           $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-                          $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>)
+                          $<INSTALL_INTERFACE:include>)
```
Comment (ruslo) [inline CMakeLists.txt]: Since GNUInstallDirs used you can do `CMAKE_INSTALL_INCLUDEDIR` here.

Comment (lsolanka) [reply CMakeLists.txt]: Done, thanks. This is relative.

File: CMakeLists.txt (hunk 1 above)
Comment (lsolanka) [reply CMakeLists.txt]: `CMAKE_INSTALL_LIBDIR` is causing me issues on Ubuntu 16.04 unfortunately. Agree that according to docs it should be relative but when I add:
```cmake
message(STATUS "CMAKE_INSTALL_LIBDIR: ${CMAKE_INSTALL_LIBDIR}")
```
and set `CMAKE_INSTALL_PREFIX=/usr/local`, then the output is:
```
-- CMAKE_INSTALL_LIBDIR: /usr/local/lib
```
With CMake version 3.10.0. Not sure why it deviates from the specs...

If I run `build.py <params> --clear --install` then the path is also an absolute path to the `_install/<toolchain>` directory where build.py was run, i.e. the same as above.

Comment (ruslo) [reply CMakeLists.txt]: Sounds like a CMake bug, can you try to make an update?

Comment (lsolanka) [reply CMakeLists.txt]: No problem, updated cmake to 3.11.1, the same thing, if I use `CMAKE_INSTALL_LIBDIR` in `install(EXPORT)`, tinyxml2Targets.cmake has this in one line:
```cmake
# The installation prefix configured by this project.
set(_IMPORT_PREFIX "/home/<my_path_to_tinyxml>/tinyxml2/_install/gcc-7-cxx17-pic")

# Create imported target tinyxml2
add_library(tinyxml2 SHARED IMPORTED)

set_target_properties(tinyxml2 PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)
```

Even tried downgrading cmake to 3.0.0. This one actually prints an error (when using `CMAKE_INSTALL_LIBDIR`):
```
CMake Error: install(EXPORT "tinyxml2Targets") given absolute DESTINATION "<cut-out-path>" but the export references an installation of target "tinyxml2" which has relative DESTINATION "lib"
```

Comment (ruslo) [reply CMakeLists.txt]: I think I have found the origin of the problem:
* https://github.com/hunter-packages/tinyxml2/blob/c1424ee4e1c1ed8129d9147ccc920ab92dd9ef86/CMakeLists.txt#L146-L151

Comment (lsolanka) [reply CMakeLists.txt]: Oh my...didn't notice that :-) :+1:
That's why `_INCLUDEDIR` in `target_include_directories()` was relative but this one wasn't...

So, I've removed the whole block, hope that's fine...it's a relatively small patch so shouldn't be a maintenance problem?

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/tinyxml2/releases/tag/v6.2.0-p0

**Comment by ruslo** [comment]
Another things that can be improved:
* Use `configure_package_config_file` to create `*Config.cmake` with `check_required_components`. In this case if user will accidentally specify COMPONENTS we will return error instead of silent ignore.
  * https://github.com/cgold-examples/fruits/blob/b266e5a8b0dccc250390346acde07e613562d85d/CMakeLists.txt#L98-L105
  * https://github.com/cgold-examples/fruits/blob/b266e5a8b0dccc250390346acde07e613562d85d/cmake/template/Config.cmake.in#L4

* `TINYXML2_DEBUG` used in installed public header, it make sense to make it `PUBLIC` for Debug configuration, i.e. add it by `target_compile_definitions(... PUBLIC ...)`.

* Remove mess with static/shared libs. `BUILD_SHARED_LIBS=ON` should result in building shared library and `BUILD_SHARED_LIBS=OFF` should result in building static library:
  * https://cgold.readthedocs.io/en/latest/tutorials/libraries/static-shared.html

**Pull request merged**